code/modules/mob/living/carbon/emote.dm

```diff
@@ -1,171 +1,64 @@
-			if (!muzzled)
-				..(act)
-			else
-				message = "<B>[src]</B> makes a strong noise."
-				m_type = 2
-
-		if ("gasp","gasps")
-			if (!muzzled)
-				..(act)
-			else
-				message = "<B>[src]</B> makes a weak noise."
-				m_type = 2
-
-		if ("giggle","giggles")
-			if (!muzzled)
-				..(act)
-			else
-				message = "<B>[src]</B> makes a noise."
-				m_type = 2
-
-		if ("laugh","laughs")
-			if (!muzzled)
-				..(act)
-			else
-				message = "<B>[src]</B> makes a noise."
-
-		if ("me")
-			if(!silent)
-				..()
-			return
-
-		if ("scream","screams")
-			if (!muzzled)
-				..(act)
-			else
-				message = "<B>[src]</B> makes a very loud noise."
-				m_type = 2
-
-		if ("sneeze","sneezes")
-			if (!muzzled)
-				..(act)
-			else
-				message = "<B>[src]</B> makes a strange noise."
-				m_type = 2
-
-		if ("sigh","sighs")
-			if (!muzzled)
-				..(act)
-			else
-				message = "<B>[src]</B> sighs."
-				m_type = 2
-
-		if ("snore","snores")
-			if (!muzzled)
-				..(act)
-			else
-				message = "<B>[src]</B> makes a noise."
-				m_type = 2
-
-		if ("whimper","whimpers")
-			if (!muzzled)
-				..(act)
-			else
-				message = "<B>[src]</B> makes a weak noise."
-				m_type = 2
-
-		if ("wink","winks")
-			message = "<B>[src]</B> winks."
-			m_type = 1
-
-		if ("yawn","yawns")
-			if (!muzzled)
-				..(act)
-
-		if ("help")
-			src << "Help for emotes. You can use these emotes with say \"*emote\":\n\naflap, airguitar, blink, blink_r, blush, bow-(none)/mob, burp, choke, chuckle, clap, collapse, cough, dance, deathgasp, drool, flap, frown, gasp, giggle, glare-(none)/mob, grin, jump, laugh, look, me, nod, point-atom, scream, shake, sigh, sit, smile, sneeze, sniff, snore, stare-(none)/mob, sulk, sway, tremble, twitch, twitch_s, wave, whimper, wink, yawn"
-
-		else
-			..(act)
-
-
-
-
-
-	if (message)
-		log_emote("[name]/[key] : [message]")
-
- //Hearing gasp and such every five seconds is not good emotes were not global for a reason.
- // Maybe some people are okay with that.
-
-		for(var/mob/M in dead_mob_list)
-			if(!M.client || isnewplayer(M))
-				continue //skip monkeys, leavers and new players
-			if(M.stat == DEAD && M.client && (M.client.prefs.chat_toggles & CHAT_GHOSTSIGHT) && !(M in viewers(src,null)))
-				M.show_message(message)
-
-
-		if (m_type & 1)
-			visible_message(message)
-		else if (m_type & 2)
-			audible_message(message)
+/datum/emote/living/carbon
+	mob_type_allowed_typecache = list(/mob/living/carbon)
+
+/datum/emote/living/carbon/airguitar
+	key = "airguitar"
+	message = "is strumming the air and headbanging like a safari chimp."
+	restraint_check = TRUE
+
+/datum/emote/living/carbon/blink
+	key = "blink"
+	key_third_person = "blinks"
+	message = "blinks."
+
+/datum/emote/living/carbon/blink_r
+	key = "blink_r"
+	message = "blinks rapidly."
+
+/datum/emote/living/carbon/clap
+	key = "clap"
+	key_third_person = "claps"
+	message = "claps."
+	muzzle_ignore = TRUE
+	emote_type = EMOTE_AUDIBLE
+
+/datum/emote/living/carbon/gnarl

```

Comment:
[has one for aliens aswell](https://github.com/tgstation/tgstation/blob/eace4eec148e8a366ee2fe6673f852b59a2e77b1/code/modules/mob/living/carbon/monkey/emote.dm#L18)